### PR TITLE
feat(storybook): add storybook and story

### LINKED ISSUE #18: Add Storybook to the project
### Description:
1. Add React Storybook to the project
2. Add a sample story for the Email component

### Outcome:
1. Working story for the Email component

### PR DESCRIPTION
**Pull request by niloysikdar**
### The PR covers:
1. add react and react-dom to the peerDependencies list (extra one)
2. add storybook to the project 
3. add support for `ts` in storybook
4. add a story for `Email` component

resolves #18

### CODE REVIEW

File: src/components/Email/Email.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import { Email } from './Email';
+
+export default {
+  component: Email,
+} as ComponentMeta<typeof Email>;
+
+export const MyEmail: ComponentStory<typeof Email> = () => (
+  <Email>
+    <h2>Hello World</h2>
```
Comment (agentmilindu) [inline src/components/Email/Email.stories.tsx:11]: Take the children from `args`

**Review by agentmilindu — CHANGES_REQUESTED** [review]

File: src/components/Email/Email.stories.tsx (hunk 1 above)
Comment (niloysikdar) [reply src/components/Email/Email.stories.tsx]: done

File: .storybook/main.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
```
Comment (agentmilindu) [inline .storybook/main.ts:2]: Can't we do 
```suggestion
  stories: ['../src/**/*.stories.@(js|jsx|ts|tsx|mdx)'],
```

File: src/components/Email/Email.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import { Email } from './Email';
+
+export default {
+  component: Email,
+} as ComponentMeta<typeof Email>;
+
+//“template” of how args map to rendering
+const Template: ComponentStory<typeof Email> = (args) => <Email {...args} />;
+
+export const MyEmail = Template.bind({});
```
Comment (agentmilindu) [inline src/components/Email/Email.stories.tsx:12]: Let's name the main story as `Default`? Then when we have different variations, we can define them with the variant name? 

For example. For buttons, 
1. Default
2. Secondary (  Secondary button variation )

File: src/components/Email/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export { Email } from './Email';
```
Comment (agentmilindu) [inline src/components/Email/index.ts:1]: Let's export the types also?

**Review by agentmilindu — CHANGES_REQUESTED** [review]

File: src/components/Email/index.ts (hunk 4 above)
Comment (niloysikdar) [reply src/components/Email/index.ts]: done

File: src/components/Email/Email.stories.tsx (hunk 3 above)
Comment (niloysikdar) [reply src/components/Email/Email.stories.tsx]: yeah, sounds cool

File: .storybook/main.ts (hunk 2 above)
Comment (niloysikdar) [reply .storybook/main.ts]: done

File: src/components/Email/index.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export { Email, EmailProps } from './Email';
```
Comment (agentmilindu) [inline src/components/Email/index.ts:1]: Oh! You exported the Component and the types in one export! Didn’t it complain? I thought you have to prefix with `type` to re-export types.

**Pull request merged**